onnxruntime/core/graph/featurizers_ops/featurizers_defs.cc

```diff
@@ -1313,6 +1315,98 @@ void RegisterRobustScalerFeaturizerVer1() {
+  //  Examples:
+  //      A simple example would be horizon = 1, window size = 2, and we want to take the mean.
+  //      +-----------+-------+-------------------+
+  //      | grain     | target| target_mean       |
+  //      +===========+=======+===================+
+  //      | A         | 10    | [NAN]             |
+  //      +-----------+-------+-------------------+
+  //      | A         | 4     | [10]              |
+  //      +-----------+-------+-------------------+
+  //      | A         | 6     | [7]               |
+  //      +-----------+-------+-------------------+
+  //      | A         | 11    | [5]               |
+  //      +-----------+-------+-------------------+
+  //      A more complex example would be, assuming we have horizon = 2, window size = 2, min window periods = 2, and we want the mean
+  //      +-----------+-------+-------------------+
+  //      | grain     | target| target_max        |
+  //      +===========+=======+===================+
+  //      | A         | 10    | [NAN, NAN]        |
+  //      +-----------+-------+-------------------+
+  //      | A         | 4     | [NAN, NAN]        |
+  //      +-----------+-------+-------------------+
+  //      | A         | 6     | [NAN, 7]          |
+  //      +-----------+-------+-------------------+
+  //      | A         | 11    | [7, 5]            |
+  //      +-----------+-------+-------------------+
+  //)DOC";
+
+  MS_FEATURIZERS_OPERATOR_SCHEMA(RollingWindowTransformer)
+      .SinceVersion(1)
+      .SetDomain(kMSFeaturizersDomain)
+      .Input(
+          0,
+          "State",
+          "State generated during training that is used for prediction",
+          "T0")
+      .Input(
+          1,
+          "Grains",
+          "Grains tensor of shape [R][K].",
+          "GrainT")
+      .Input(
+          2,
+          "Target",
+          "Target tensor of shape [R]",
+          "T")
+      .Output(
+          0,
+          "Output",
+          "Output tensor of shape [R][M]",
+          "OutputT")
+      .TypeConstraint(
+          "T0",
+          {"tensor(uint8)"},
+          "No information is available")
+      .TypeConstraint(
+          "GrainT",
+          {"tensor(string)"},
+          "No information is available")
+      .TypeConstraint(
+          "T",
+          {"tensor(int8)", "tensor(uint8)", "tensor(int16)",  "tensor(uint16)", "tensor(int32)", "tensor(uint32)", "tensor(int64)", "tensor(uint64)", "tensor(float)", "tensor(double)"},
+          "No information is available")
+      .TypeConstraint(
+          "OutputT",
+          {"tensor(double)"},
+          "No information is available")
+      .TypeAndShapeInferenceFunction(
+          [](ONNX_NAMESPACE::InferenceContext& ctx) {
+            propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_DOUBLE, 0);
+            if (hasInputShape(ctx, 1) && hasInputShape(ctx, 2)) {
```

Comment:
>f (hasInputShape(ctx, 1) && hasInputShape(ctx, 2)) { [](start = 13, length = 52)

You should not fail inference if any of the shapes is not available. The output shape depends on the grains_shape but if it is not available do not fail the function. Also if the target shape is not available do not fail. Fail only if it available but the dims are not correct.

Reply:
I think I fail it when shape is available but dims are not correct